node/pkg/p2p/p2p.go

```diff
@@ -280,7 +280,9 @@ func Run(
 			ourAddr := ethcrypto.PubkeyToAddress(gk.PublicKey)
 
 			ctr := int64(0)
-			timer := time.NewTimer(time.Nanosecond) // Guardians should send out their first heartbeat immediately to speed up test runs.
+			// Guardians should send out their first heartbeat immediately to speed up test runs.
+			// But we also want to wait a little bit such that network connections can be established by then.
+			timer := time.NewTimer(time.Second * 2)
```

Comment:
Is there a way to actually verify the connections are successful? Any way other than a hardcoded sleep?

Reply:
Even if the first heartbeat after 2s is not received by the other guardians, one of the next heartbeats (in 15s intervals) will eventually be received. In other words, we don't depend on the 2 seconds to be sufficient time, it's just a good value because it makes the tests run faster in practice.

Reply:
2 seconds is just less time. I meant making no timer and just verifying the connection is successful, but this is an improvement. It's just improving a sloppy approach to a faster still sloppy approach (not your fault)

Reply:
afaik you can force the connection (which we're doing already), but there's no way to verify that gossip is actually connected, so this was the best I could do.